src/ArcxAnalyticsSdk.ts

```diff
@@ -17,32 +18,74 @@ export class ArcxAnalyticsSdk {
     public readonly identityId: string,
     private readonly sdkConfig: SdkConfig,
   ) {
+    if (sdkConfig.trackPages || sdkConfig.trackReferrer || sdkConfig.trackUTM) {
+      this._trackFirstPageVisit()
+    }
+
     if (this.sdkConfig.trackPages) {
       this.trackPagesChanges()
     }
-    if (this.sdkConfig.trackReferrer) {
-      this.referrer()
-    }
   }
 
   /**********************/
   /** INTERNAL METHODS **/
   /**********************/
 
-  private trackPagesChanges() {
-    const currentUrl = sessionStorage.getItem(CURRENT_URL_KEY)
-    if (!currentUrl) {
-      sessionStorage.setItem(CURRENT_URL_KEY, location.href)
-      this.page({ url: location.href })
+  private _trackFirstPageVisit() {
+    // If no tracking is set, this function shouldn't have been called
+    if (!this.sdkConfig.trackPages && !this.sdkConfig.trackReferrer && !this.sdkConfig.trackUTM) {
+      throw new Error('ArcxAnalyticsSdk::_trackFirstPageVisit: No tracking is set')
     }
 
+    const attributes: {
+      url?: string
+      referrer?: string
+      utm?: {
+        source?: string
+        medium?: string
+        campaign?: string
+      }
```

Comment:
@stuart-watt this would be the attribute structure of the `FIRST_PAGE_VISIT` event. Would you like to change anything?

Reply:
This is perfect for me, no changes required @gtupak

Reply:
It would actually be simpler on my end if the `url`, `referrer`, `source`, `medium` and `campaign` were all on the same level. i.e. no need for the `utm` key. Since I know what `source`, `medium` and `campaign` are UTM tags, I don't see a need to explicitly have a nested utm dictionary.

Reply:
I find it better structured this way. In case we want to add more information to this call in the future, we can add other nested objects inside based on their category. If we put all of them at the same level it might create confusion in the future and increase tech debt potential.